### DISCUSSION TITLE
Transform class not working properly on Safari

### DISCUSSION THREAD

**Discussion by stefan-krajnik** [tailwindlabs/tailwindcss#8788] [Help]
**What version of Tailwind CSS are you using?**

3.1.0

**What build tool (or framework if it abstracts the build tool) are you using?**

postcss 8.4.14, Next.js 16+, webpack 5.73.0

**What version of Node.js are you using?**

v16+

**What browser are you using?**

Safari 13.1.3

**What operating system are you using?**

macOS

**Describe your issue**

I noticed the problem when found out `rotate-90` / `-rotate-90` were not working.  
The class works properly in other browsers but not in Safari 13.1.3 ,

It seems like default css vars are not working and that's causing the problem.

<img width="299" alt="Screenshot 2022-07-06 at 06 57 52" src="https://user-images.githubusercontent.com/16229997/177535293-d7631f0a-1623-48bc-a1b4-fc91d5b5cbe3.png">
 
 
If I edit the class I can make it work , e.g.
_Notice the `0` fallback for translate Y , it doesn't work without the fallback._
 
<img width="298" alt="Screenshot 2022-07-06 at 07 01 07" src="https://user-images.githubusercontent.com/16229997/177535915-8fd699fd-78e9-4800-95aa-824894c03270.png">

**Comment by K3NZ11** [comment]
@stefan-krajnik It seems to be working fine for me. Have you tested this issue on the latest version (`v3.1.4`) of Tailwind?

**Comment by adamwathan** [comment]
Testing this on Safari 13.1 and it does seem to work as expected for me:

https://jsfiddle.net/e83sp7Lz/

<img width="1498" alt="image" src="https://user-images.githubusercontent.com/4323180/177584560-8665ce6f-256a-4c6d-942d-1f2326891de7.png">

Can you provide a reproduction project that actually demonstrates it breaking? We don't really aim to support Safari < 14 but this should still work.

**Reply by mattnicholson** [reply]
Any ideas why Safari would be working correctly on your demo there, and on the Tailwind website, but not on my local set up? It looks as if it's ignoring the default variables established in *,::after,::before so the transforms are failing due to missing variables. Looks good on the tailwind site though and I'm using 3.2.4. I can see the stylesheet has generated those variables, but they don't show as inherited values in the web inspector?

**Comment by stefan-krajnik** [comment]
@K3NZ11 @adamwathan thanks for your help

I finally found the reason why it wasn't working. The problem has been fixed already in https://github.com/tailwindlabs/tailwindcss/releases/tag/v3.1.1 ( https://github.com/tailwindlabs/tailwindcss/pull/8567 ) 

The problem was here

```css
*, ::before, ::after, ::backdrop {
  --tw-border-spacing-x: 0;
.......
}
```

it was caused by `::backdrop`, when dropped it worked properly

I updated to the latest version and everything works as expected

**Reply by csellis** [reply]
How did you drop it? I'm on 3.1.6 and still experiencing

**Comment by mdoyle13** [comment]
In my case (using tailwind 3.2.4) w/ Safari 14 - the issue with `translate` utility classes not working only happens when the output is minified. Which is why it works in the playground and not when its built in my project.

I'm using the rails gem which wraps the standalone CLI so it could be an issue there as well.

For now I'm downgrading to tw 3.2.1 (rails gem version 2.0.17) which works with minifcation on for my usecases.

**Reply by marciscaupjonoks** [reply]
I experienced the same thing on 3.3.2 and iOS, v14.5 - fixed it by removing "--minify" from the build command.

**Reply by abinoda** [reply]
I'm running into this same issue and just opened an issue: https://github.com/tailwindlabs/tailwindcss/issues/12160
